pkg/runit/servicebuilder.go

```diff
@@ -83,13 +85,15 @@ __END__
 	return []byte(ret), nil
 }
 
-func (s ServiceTemplate) LogScript() ([]byte, error) {
+func (s ServiceTemplate) logScript() ([]byte, error) {
 	sleep := 2
 	if s.LogSleep != nil && *s.LogSleep >= 0 {
 		sleep = *s.LogSleep
 	}
 
 	log := s.Log
+	// TODO this leaks the type information :<
+	// should we implement empty? on this type
```

Comment:
The line below is a bit more disconcerting wrt type abstraction. I'm not sure this TODO is worth addressing - we're getting a lot of benefits from the type checks at function boundaries, adding any more feels like contortionism.